extensions/liquibase/runtime/src/main/java/io/quarkus/liquibase/QLiquibase.java

```diff
@@ -0,0 +1,932 @@
+package io.quarkus.liquibase;
+
+import java.io.PrintStream;
+import java.io.Writer;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.Collection;
+import java.util.Date;
+import java.util.List;
+
+import javax.sql.DataSource;
+
+import io.quarkus.liquibase.runtime.LiquibaseConfig;
+import liquibase.CatalogAndSchema;
+import liquibase.Contexts;
+import liquibase.LabelExpression;
+import liquibase.Liquibase;
+import liquibase.change.CheckSum;
+import liquibase.changelog.ChangeSet;
+import liquibase.changelog.ChangeSetStatus;
+import liquibase.changelog.DatabaseChangeLog;
+import liquibase.changelog.RanChangeSet;
+import liquibase.database.Database;
+import liquibase.database.DatabaseFactory;
+import liquibase.database.jvm.JdbcConnection;
+import liquibase.diff.DiffResult;
+import liquibase.diff.compare.CompareControl;
+import liquibase.exception.DatabaseException;
+import liquibase.exception.LiquibaseException;
+import liquibase.lockservice.DatabaseChangeLogLock;
+import liquibase.resource.ResourceAccessor;
+
+/**
+ * The quarkus liquibase bean
+ */
+public class QLiquibase {
```

Comment:
Why `Q`? It doesn't sound very natural to me. Especially for an API.

Why do we need this wrapper? In the Flyway extension, we inject the `Flyway` instance which seems more future proof.

Reply:
I am really not sure if the QLiquibase (wrapper class) is the right name. 

I started like the Flyway:
- The Liquibase is expecting open JDBC connection in the constructor (it reads the info about the database from the connection). In the case of Flyway we have Datasource as input parameter
- The Liquibase does not open or close the connection
- I tried to extend the Liquibase object and overwrite the method to open and close the connection but there are final methods like dropAll()

They have CDI integration: https://github.com/liquibase/liquibase/blob/628a95e7344ad2bc108746fd3491e3957879668e/liquibase-cdi/src/main/java/liquibase/integration/cdi/CDILiquibase.java#L130

If we use the Liquibase object in the producer we can open connection and the developer needs to close the connection.
If we use the extended Liquibase implementation in the producer we need to mark the final methods which do not close the connection. 
If we use the wrapper (with correct name) we can open and close the connection for the developer for each method but we have much more code.

The QLiquibase also contains the ```LiquibaseConfig``` object which is merge of the run-time configuration and build configuration.

I am open to any solution

Reply:
I renamed the QLiquibase class in Liquibase